sourcegraph/sourcegraph
# Migrate search export to the core workflow

Migrates [search-export extension](https://github.com/sourcegraph/sourcegraph-search-export) functionality into the core workflow.
Closes https://github.com/sourcegraph/sourcegraph/issues/38379



**Notes on implementation**
1. This PR migrates the existing extension functionality with known limitations (we do not actually export all search results found, but only a certain portion of it, what actually the [`query SearchResults` returns](https://github.com/sourcegraph/sourcegraph-search-export/blob/master/src/extension.ts#L73)).
2. There are [2 extension specific user settings](https://github.com/sourcegraph/sourcegraph-search-export/blob/master/src/extension.ts#L38-L39):  `searchExport.searchPatternType` and `searchExport.maxMatchContentLength`. We migrate the former but ignore the latter as it targets Sourcegraph versions [prior to 3.29](https://github.com/sourcegraph/sourcegraph-search-export/blob/master/src/extension.ts#L50-L62), and the migrated functionality will be available only for 4.0+ Sourcegraph users.
3. The only visual change in the regular UI is the export button text; notification behavior remains the same.
4. Export search results button is added to the simple UI actions dropdown.

| Extension | Core workflow feature |
| -- | -- |
|  <img width="2541" alt="Screenshot 2022-08-02 at 17 54 03" src="https://user-images.githubusercontent.com/25318659/182420562-ebde9077-c8ee-4e20-8690-e8811f75dbae.png"><img width="2549" alt="Screenshot 2022-08-02 at 17 53 22" src="https://user-images.githubusercontent.com/25318659/182420739-c2290172-429a-422a-bd51-2aeae1776c75.png"> | <img width="2543" alt="Screenshot 2022-08-02 at 18 22 03" src="https://user-images.githubusercontent.com/25318659/182420494-4f3502ee-38b8-4af7-97e7-649116ca4283.png"> <img width="2549" alt="Screenshot 2022-08-02 at 17 53 22" src="https://user-images.githubusercontent.com/25318659/182420739-c2290172-429a-422a-bd51-2aeae1776c75.png">|
||<img width="418" alt="Screenshot 2022-08-03 at 15 19 27" src="https://user-images.githubusercontent.com/25318659/182606713-42df034f-ad33-4969-8173-8e65caf56eea.png">|

## Test plan
1. `sg start web-standalone`
5. Enable `sourcegraph/search-export` extension
6. Search for something, click "Export search results" button and download the suggested file
7. Enable `extensionsAsCoreFeatures` experimental feature in user settings.
8. Repeat step 3 and ensure the same search results are downloaded.



## App preview:

- [Web](https://sg-web-taras-yemets-migrate-search-export.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-axsvitqpee.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
